### PR TITLE
Adjust for Django 1.8's handling of TEMPLATES/TEMPLATE_CONTEXT_PROCES…

### PR DESCRIPTION
**Pull request by spookylukey**
…SORS in docs/config checks

See https://docs.djangoproject.com/en/dev/ref/templates/upgrading/#the-templates-settings for more info.

### CODE REVIEW

**Comment by benjaoming** [comment]
This fixes #424

File: wiki/models/__init__.py [hunk 1]
```diff
@@ -42,13 +42,21 @@
...
         'django-wiki: needs django.contrib.sites in INSTALLED_APPS')
 
+TEMPLATE_CONTEXT_PROCESSORS = []
+if hasattr(django_settings, 'TEMPLATES'):
+    # Django 1.8 compat
+    backends = [b for b in django_settings.TEMPLATES if b.get('BACKEND', '') == 'django.template.backends.django.DjangoTemplates']
+    if len(backends) == 1:
+        TEMPLATE_CONTEXT_PROCESSORS = backends[0].get('OPTIONS', {}).get('context_processors', [])
+
 if not 'django.contrib.auth.context_processors.auth' in django_settings.TEMPLATE_CONTEXT_PROCESSORS:
```
Comment (benjaoming) [inline wiki/models/__init__.py]: Shouldn't this be checking the new `TEMPLATE_CONTEXT_PROCESSORS` and shouldn't there be an `else` block for pre-1.8 versions that goes

    else:
        TEMPLATE_CONTEXT_PROCESSORS = django_settings.TEMPLATE_CONTEXT_PROCESSORS

**Comment by benjaoming** [comment]
Great, thanks for fixing this!! So I take it that you are working on a project that's quite good at keeping itself up to date with new versions of Django? Would love to hear about your experiences with this one day...

Have a look at the comment, would you agree on this?

File: wiki/models/__init__.py (hunk 1 above)
Comment (spookylukey) [inline wiki/models/__init__.py]: You're right, my change doesn't work correctly for older projects. I'll submit a fix.

Comment (benjaoming) [inline wiki/models/__init__.py]: most wonderful! thanks!

**Comment by benjaoming** [comment]
Great work @spookylukey ! Thanks!

**Comment by emorozov** [comment]
I think this doesn't work still. Shouldn't this line:
```
if not 'django.contrib.auth.context_processors.auth' in django_settings.TEMPLATE_CONTEXT_PROCESSORS:
```
be:
```
if not 'django.contrib.auth.context_processors.auth' in TEMPLATE_CONTEXT_PROCESSORS:
```

I still get the error on Django 1.9.9

**Comment by y2kbadbug** [comment]
I agree, that this is still broken:
``
Traceback (most recent call last):
  File "./manage.py", line 10, in <module>
    execute_from_command_line(sys.argv)
  File "/home/dshafer/django/lib/python2.7/site-packages/django/core/management/__init__.py", line 367, in execute_from_command_line
    utility.execute()
  File "/home/dshafer/django/lib/python2.7/site-packages/django/core/management/__init__.py", line 341, in execute
    django.setup()
  File "/home/dshafer/django/lib/python2.7/site-packages/django/__init__.py", line 27, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/home/dshafer/django/lib/python2.7/site-packages/django/apps/registry.py", line 108, in populate
    app_config.import_models(all_models)
  File "/home/dshafer/django/lib/python2.7/site-packages/django/apps/config.py", line 199, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/lib64/python2.7/importlib/__init__.py", line 37, in import_module
    __import__(name)
  File "/home/dshafer/django/lib/python2.7/site-packages/wiki/models/__init__.py", line 54, in <module>
    if 'django.contrib.auth.context_processors.auth' not in django_settings.TEMPLATE_CONTEXT_PROCESSORS:
  File "/home/dshafer/django/lib/python2.7/site-packages/django/conf/__init__.py", line 54, in __getattr__
    return getattr(self._wrapped, name)
AttributeError: 'Settings' object has no attribute 'TEMPLATE_CONTEXT_PROCESSORS'
``

 I'm on Django 1.10.4

**Comment by benjaoming** [comment]
@y2kbadbug - it's fixed in the current master and 0.2b2 prerelease

https://github.com/django-wiki/django-wiki/commit/d9d05f63bde76dbc957e7c2032d5c71aa7e4d62a

**Comment by benjaoming** [comment]
@y2kbadbug - it's fixed in the current master and 0.2b2 prerelease

https://github.com/django-wiki/django-wiki/commit/d9d05f63bde76dbc957e7c2032d5c71aa7e4d62a

**Pull request merged**